src/Interpreters/Aggregator.h

```diff
@@ -1224,6 +1231,28 @@ class Aggregator final
         bool final,
         ThreadPool * thread_pool) const;
 
+    template <typename Method, typename Table>
+    void convertToBlockImplSingleKey(
+        Method & method,
+        Table & data,
+        MutableColumns & key_columns,
+        AggregateColumnsData & aggregate_columns,
+        MutableColumns & final_aggregate_columns,
+        Arena * arena,
+        const std::map<String, Field> & key) const;
+
+    template <typename Method>
+    Block convertOneBucketToBlockSingleKey(
+        AggregatedDataVariants & data_variants,
+        Method & method,
+        Arena * arena,
+        size_t bucket,
+        const std::map<String, Field> & key) const;
+
+    Block mergeAndFillBlockForSingleKey(
```

Comment:
Comment is needed.
Why do we use `std::map<String, Field>` for key?

Reply:
It's temporal and I plan to replace it with ColumnRawPtrs very soon.

Reply:
Ok, I see.
So, `key` represents a single row. We fill it if there is such a key in state.